File: frontend/src/api/middleware.ts
import http from './_http';

export default {
  index (workspaceId: number) {
    return http.get(`/workspaces/${workspaceId}/middlewares/`);
  },
  create (workspaceId: number, data: any) {
    return http.post(`/workspaces/${workspaceId}/middlewares/`, data);
  },
  update (workspaceId: number, id: number, data: any) {
    return http.put(`/workspaces/${workspaceId}/middlewares/${id}`, data);
  },
  delete (workspaceId: number, id: number) {
    return http.delete(`/workspaces/${workspaceId}/middlewares/${id}`);
  },
}

File: backend/routes/middleware.go
package routes

import (
	"app/controllers/middleware"
	"app/middlewares"

	"github.com/gin-gonic/gin"
)

func InitMiddleware(r *gin.RouterGroup, e *gin.Engine) {
	const baseUrl = "/:id/middlewares"

	r.GET(baseUrl + "/", middlewares.CheckWorkspace, middleware.Index)
	r.POST(baseUrl + "/", middlewares.CheckWorkspace, middleware.Create)
	r.PUT(baseUrl + "/:middlewareId", middlewares.CheckWorkspace, middleware.Update)
	r.DELETE(baseUrl + "/:middlewareId", middlewares.CheckWorkspace, middleware.Delete)
}
